google/cloud/bigtable/mutations_batcher.py

```diff
@@ -15,16 +15,117 @@
 from __future__ import annotations
 
 import asyncio
+import atexit
+import warnings
 from typing import TYPE_CHECKING
 
-from google.cloud.bigtable.mutations import Mutation
-from google.cloud.bigtable.row_filters import RowFilter
+from google.cloud.bigtable.mutations import RowMutationEntry
+from google.cloud.bigtable.exceptions import MutationsExceptionGroup
+from google.cloud.bigtable.exceptions import FailedMutationEntryError
+
+from google.cloud.bigtable._mutate_rows import _mutate_rows_operation
 
 if TYPE_CHECKING:
     from google.cloud.bigtable.client import Table  # pragma: no cover
 
-# Type alias used internally for readability.
-_row_key_type = bytes
+# used to make more readable default values
+MB_SIZE = 1024 * 1024
+
+
+class _FlowControl:
+    """
+    Manages flow control for batched mutations. Mutations are registered against
+    the FlowControl object before being sent, which will block if size or count
+    limits have reached capacity. When a mutation is complete, it is unregistered
+    from the FlowControl object, which will notify any blocked requests that there
+    is additional capacity.
+    """
+
+    def __init__(self, max_mutation_count: int | None, max_mutation_bytes: int | None):
+        """
+        Args:
+          - max_mutation_count: maximum number of mutations to send in a single rpc.
+             This corresponds to individual mutations in a single RowMutationEntry.
+             If None, no limit is enforced.
+          - max_mutation_bytes: maximum number of bytes to send in a single rpc.
+             If None, no limit is enforced.
```

Comment:
should the args be flow_control_max_count, flow_control_max_bytes?

Reply:
I thought that could be left out since this is the flow control class, and they will be accessed as "flow_control.max_mutation_count". But I can change it to be more explicit if you'd prefer